lusSTR/filter_settings.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env python
+#
+# -----------------------------------------------------------------------------
+# Copyright (c) 2020, Battelle National Biodefense Institute.
+#
+# This file is part of lusSTR (http://github.com/bioforensics/lusSTR)
+# and is licensed under the BSD license: see LICENSE.txt.
+# -----------------------------------------------------------------------------
+
+import json
+import lusSTR
+import numpy as np
+import pandas as pd
+from pkg_resources import resource_filename
+
+
+def get_filter_metadata_file():
+    return resource_filename('lusSTR', 'filters.json')
+
+
+with open(get_filter_metadata_file(), 'r') as fh:
+    filter_marker_data = json.load(fh)
+
+
+def filters(data_order, locus, total_reads):
+    metadata = filter_marker_data[locus]
+    if len(data_order) == 1:
+        if thresholds('Detection', metadata, total_reads, data_order['Reads'][0])[1] is False:
+            data_order = pd.DataFrame()
+        elif thresholds('Analytical', metadata, total_reads, data_order['Reads'][0])[1] is False:
+            data_order[['allele_type', 'perc_noise']] = ['noise', 1.0]
+        elif thresholds('Analytical', metadata, total_reads, data_order['Reads'][0])[1] is True:
+            data_order['allele_type'] = 'real_allele'
```

Comment:
This function is...high maintenance. :smiley:

On one hand, the tests are passing, so I don't want to hold up progress on this PR too much. On the other hand, I worry about adding support for filtering LUS/LUS+ and SNP alleles with this as the foundation.

I've often been ok with not being able to grasp all the details when doing code reviews for lusSTR. But if you're agreeable, perhaps we can schedule some time to walk through this code in person (or in real time remotely) next week.